text/0080-datetime-extension.md

```diff
@@ -0,0 +1,488 @@
+
+**Prevent employees from accessing work documents on the weekend.**
+
+With Unix Time, this requires `/` and `%` operators to compute the `dayOfWeek`, which isn't currently expressible in Cedar.
+
+**Permit access to a special opportunity for persons born on Leap Day.**
+
+With Unix Time, this requires `/` and `%` operators to compute whether or not it is a leap year, which isn't currently expressible in Cedar.
+
+**Forbid access to EU resources after Brexit**
+
+```cedar
+forbid(
+  principal,
+  action,
+  resource
+) when {
+  context.currentTime > 1580511600 &&
+  context.location.countryOfOrigin == 'GB' &&
+  resource.owner == 'EU'
+}
+```
+
+### Alt. B: Pass results of time checks in the context
+
+Another workaround we have suggested is to simply handle date/time logic _outside_ of Cedar, and pass the results of checks in the `context`. For example, you could pass in fields like  `context.isWorkingHours` or `context.dayOfTheWeek`.
+
+Here are the previous examples rewritten to use additional context.
+
+**Only allow experienced, tenured persons from the Hardware Engineering department to see prototypes.**
+
+```cedar
+permit(
+  principal is User,
+  action == Action::"view",
+  resource in Folder::"device_prototypes"
+)
+when {
+  principal.department == "HardwareEngineering" &&
+  principal.jobLevel >= 10 &&
+  principal.hireDate <= context.minimumHiringDateForAccess
+};
+```
+
+Note: assumes `context.hireDate` and `context.minimumHiringDateForAccess` are `long` values (e.g., Unix time).
+
+**Only allow user "alice" to view JPEG photos for one week after creation.**
+
+```cedar
+permit(
+  principal == User::"alice",
+  action == PhotoOp::"view",
+  resource is Photo
+) when {
+  resource.fileType == "JPEG" &&
+  resource.creationDate >= context.oldestViewableDate
+};
+```
+
+Note: assumes `resource.creationDate` and `context.oldestViewableDate` are `long` values (e.g., Unix time).
+
+**Allow access from a certain IP address only between 9am and 6pm UTC.**
+
+```cedar
+permit(
+  principal,
+  action == Action::"access",
+  resource
+) when {
+  context.srcIp.isInRange(ip("192.168.1.0/24")) &&
+  context.isWorkingHours
+};
+```
+
+**Prevent employees from accessing work documents on the weekend.**
+
+```cedar
+forbid(
+  principal,
+  action == Action::"access",
+  resource is Document
+) when {
+  context.isTheWeekend
+};
+```
+
+**Permit access to a special opportunity for persons born on Leap Day.**
+
+```cedar
+permit(
+  principal,
+  action == Action::"redeem",
+  resource is Prize
+) when {
+  context.isMyBirthday && context.isLeapDay
+};
+```
+
+**Forbid access to EU resources after Brexit**
+
+```cedar
+forbid(
+  principal,
+  action,
+  resource
+) when {
+  context.afterBrexit &&
+  context.location.countryOfOrigin == 'GB' &&
+  resource.owner == 'EU'
+}
+```
+
+## Unresolved questions
```

Comment:
Are these really "unresolved"? Maybe rename to "discussed questions" as a nicely formatted summary of the discussion? Might also be nice to summarize other topics here.